lib/renderer/chrome-api.ts

```diff
@@ -5,8 +5,30 @@ import * as url from 'url'
 // Todo: Import once extensions have been turned into TypeScript
 const Event = require('@electron/internal/renderer/extensions/event')
 
+const ProxyTabProperty = <G extends keyof Electron.WebContents>(getMethod: G, mapGet?: (val: any) => any) => (target: Tab, propertyKey: keyof chrome.tabs.Tab) => {
+  Object.defineProperty(target, propertyKey, {
+    get: function (this: Tab): any {
+      const val = ipcRendererUtils.invokeSync('CHROME_TABS_PROPERTY_GET', this.id, getMethod)
+      return mapGet ? mapGet(val) : val
+    },
+    configurable: false,
+    enumerable: true
+  })
+}
+
 class Tab {
   public id: number
+  public index = 0
+  public highlighted = false
+  @ProxyTabProperty('isFocused') public active!: boolean
+  public pinned = false
+  @ProxyTabProperty('isCurrentlyAudible') public audible!: boolean
+  @ProxyTabProperty('isDestroyed') public discarded!: boolean
+  public autoDiscardable = false
+  @ProxyTabProperty('getURL') public url!: string
+  @ProxyTabProperty('getTitle') public title!: string
+  @ProxyTabProperty('isLoadingMainFrame', (loading: boolean) => loading ? 'loading' : 'complete') public status!: string
+  public incognito = false
```

Comment:
Would it be worthwhile to implement the rest of the [`Tab`](https://developer.chrome.com/extensions/tabs#type-Tab) properties with placeholders? `windowId`, `width`, `height`, etc.